rfcs/0139-declarative-filesystem-layouts.md

```diff
@@ -0,0 +1,66 @@
+---
+feature: declarative-filesystem-layouts
+start-date: 2023-01-11
+author: l0b0
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: https://github.com/NixOS/nixpkgs/issues/209988 (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+One of the hardest parts of setting up a NixOS (or any Linux) system from scratch is setting up the filesystems. Deciding how big each partition needs to be, making sure it's UEFI compliant, whether to use LUKS inside LVM or LVM inside LUKS, remembering to set a partition bootable, etc. It would be fantastic to have a declarative way to deal with this. Features which come to mind include:
+
+- Integrates with the GUI installer. Point the installer to a `configuration.nix` with a disk layout somewhere, and it does all the necessary setup, asking for things like passphrases where necessary.
```

Comment:
If this does get accepted I think it should happen _after_ a revamp to the Calamares installer, which from what I've seen (and heard as anecdotal evidence) is quite lacking and prone to not installing properly.